platforms/android/library-compose/src/androidTest/java/io/element/android/wysiwyg/compose/RichTextEditorStateTest.kt

```diff
@@ -54,6 +57,37 @@ class RichTextEditorStateTest {
         onView(withText("Hello, world")).check(matches(isDisplayed()))
     }
 
+    @Test
+    fun testStateRestoration() = runTest {
+        val state = RichTextEditorState()
+        val hideEditor = MutableStateFlow(false)
+        composeTestRule.setContent {
+            MaterialTheme {
+                val hide by hideEditor.collectAsState()
+                Column() {
+                    if (!hide) {
+                        Text("Editor")
+                        RichTextEditor(modifier = Modifier.fillMaxWidth(), state = state)
+                    }
+                }
+            }
+        }
+
+        state.setHtml("Hello, world")
+        // Ensure line count is set
+        Assert.assertEquals(1, state.lineCount)
+
+        // Hide and show the editor to simulate a configuration change
+        hideEditor.emit(true)
+        hideEditor.emit(false)
+        composeTestRule.awaitIdle()
```

Comment:
Does this not need an extra `composeTestRule.awaitIdle()` between hiding and showing?

Reply:
I just added it and it makes no difference so I can leave it there if it helps ensure we're actually changing the state.